crates/flox/src/commands/channel.rs

```diff
@@ -76,122 +88,131 @@ pub struct Search {
 impl Search {
     pub async fn handle(self, flox: Flox) -> Result<()> {
         subcommand_metric!("search");
-        let channels = flox
-            .channels
-            .iter()
-            .filter_map(|entry| {
-                if HIDDEN_CHANNELS.contains_key(&*entry.from.id) {
-                    None
-                } else if DEFAULT_CHANNELS.contains_key(&*entry.from.id) {
-                    Some((ChannelType::Flox, entry))
-                } else {
-                    Some((ChannelType::User, entry))
-                }
-            })
-            .sorted_by(|a, b| Ord::cmp(a, b));
 
-        // Create `registry` parameter for `pkgdb`
-        let mut priority: Vec<String> = Vec::new();
-        let mut inputs = serde_json::Map::new();
-        for (_, entry) in channels {
-            // TODO: handle `subtrees` and `stabilities`
-            priority.push(entry.from.id.to_string());
-            inputs.insert(
-                entry.from.id.to_string(),
-                json!({"from": entry.from.to_string()}),
-            );
+        if self.search_term.is_none() {
+            bail!(
+                "'flox search' requires at least one search term, run 'flox search -h' for more information."
+            )
         }
```

Comment:
should we just make the argument required then?
`flox search EOL` used to be an alias for search --all, there should be no more reason for that search term to be an option

line 75, this file: https://github.com/flox/flox/pull/273/files#diff-96ed01fcdb72d1e1fb2617335124bda23c641c28444cdb5fa24a1ef2ab864142R75

Reply:
Yes we should, I wasn't sure if it being optional was required by non-feature-flagged behavior. I'm still a little unclear on where the non-flagged behavior works.

Reply:
Fixed